File: npm/src/identity/preset.js
'use strict'

// knownAgentIds retorna a lista canônica de ids de agentes conhecidos pelo
// trackfw, em ordem estável e determinística. Espelha
// internal/identity/preset.go:KnownAgentIDs.
function knownAgentIds() {
  return [
    'architect',
    'backend',
    'frontend',
    'qa',
    'infra',
    'security',
    'dba',
    'ux',
    'code-quality',
    'data',
    'iac',
    'tooling',
  ]
}

// presets contém o catálogo de presets temáticos de identidade. display_name
// e slug são HARDCODED aqui — não derivados chamando slugify em tempo de
// execução — porque essa é uma decisão explícita do ADR: hardcoding evita
// depender do comportamento de normalização Unicode sendo idêntico entre os
// CLIs Go, Node.js e Python. Espelha internal/identity/preset.go:presets.
const PRESETS = {
  greek: {
    architect: { display_name: 'Zeus', slug: 'zeus' },
    backend: { display_name: 'Apolo', slug: 'apolo' },
    frontend: { display_name: 'Afrodite', slug: 'afrodite' },
    qa: { display_name: 'Ártemis', slug: 'artemis' },
    infra: { display_name: 'Ares', slug: 'ares' },
    security: { display_name: 'Hades', slug: 'hades' },
    dba: { display_name: 'Poseidon', slug: 'poseidon' },
    ux: { display_name: 'Atena', slug: 'atena' },
    'code-quality': { display_name: 'Hefesto', slug: 'hefesto' },
    data: { display_name: 'Métis', slug: 'metis' },
    iac: { display_name: 'Dédalo', slug: 'dedalo' },
    tooling: { display_name: 'Prometeu', slug: 'prometeu' },
  },
  norse: {
    architect: { display_name: 'Odin', slug: 'odin' },
    backend: { display_name: 'Thor', slug: 'thor' },
    frontend: { display_name: 'Freya', slug: 'freya' },
    qa: { display_name: 'Heimdall', slug: 'heimdall' },
    infra: { display_name: 'Tyr', slug: 'tyr' },
    security: { display_name: 'Vidar', slug: 'vidar' },
    dba: { display_name: 'Njord', slug: 'njord' },
    ux: { display_name: 'Idun', slug: 'idun' },
    'code-quality': { display_name: 'Bragi', slug: 'bragi' },
    data: { display_name: 'Mimir', slug: 'mimir' },
    iac: { display_name: 'Ivaldi', slug: 'ivaldi' },
    tooling: { display_name: 'Loki', slug: 'loki' },
  },
  potter: {
    architect: { display_name: 'Dumbledore', slug: 'dumbledore' },
    backend: { display_name: 'Snape', slug: 'snape' },
    frontend: { display_name: 'Luna', slug: 'luna' },
    qa: { display_name: 'Moody', slug: 'moody' },
    infra: { display_name: 'Hagrid', slug: 'hagrid' },
    security: { display_name: 'Kingsley', slug: 'kingsley' },
    dba: { display_name: 'Flitwick', slug: 'flitwick' },
    ux: { display_name: 'Tonks', slug: 'tonks' },
    'code-quality': { display_name: 'Hermione', slug: 'hermione' },
    data: { display_name: 'Trelawney', slug: 'trelawney' },
    iac: { display_name: 'Rowena', slug: 'rowena' },
    tooling: { display_name: 'Ollivander', slug: 'ollivander' },
  },
  thrones: {
    architect: { display_name: 'Tyrion', slug: 'tyrion' },
    backend: { display_name: 'Jon', slug: 'jon' },
    frontend: { display_name: 'Sansa', slug: 'sansa' },
    qa: { display_name: 'Arya', slug: 'arya' },
    infra: { display_name: 'Brienne', slug: 'brienne' },
    security: { display_name: 'Varys', slug: 'varys' },
    dba: { display_name: 'Samwell', slug: 'samwell' },
    ux: { display_name: 'Margaery', slug: 'margaery' },
    'code-quality': { display_name: 'Stannis', slug: 'stannis' },
    data: { display_name: 'Bran', slug: 'bran' },
    iac: { display_name: 'Gendry', slug: 'gendry' },
    tooling: { display_name: 'Qyburn', slug: 'qyburn' },
  },
  chaves: {
    architect: { display_name: 'Girafales', slug: 'girafales' },
    backend: { display_name: 'Madruga', slug: 'madruga' },
    frontend: { display_name: 'Chiquinha', slug: 'chiquinha' },
    qa: { display_name: 'Florinda', slug: 'florinda' },
    infra: { display_name: 'Barriga', slug: 'barriga' },
    security: { display_name: 'Quico', slug: 'quico' },
    dba: { display_name: 'Clotilde', slug: 'clotilde' },
    ux: { display_name: 'Popis', slug: 'popis' },
    'code-quality': { display_name: 'Nhonho', slug: 'nhonho' },
    data: { display_name: 'Godinez', slug: 'godinez' },
    iac: { display_name: 'Chaves', slug: 'chaves' },
    tooling: { display_name: 'Chapolin', slug: 'chapolin' },
  },
  pioneers: {
    architect: { display_name: 'Turing', slug: 'turing' },
    backend: { display_name: 'Ritchie', slug: 'ritchie' },
    frontend: { display_name: 'Berners-Lee', slug: 'berners-lee' },
    qa: { display_name: 'Hamilton', slug: 'hamilton' },
    infra: { display_name: 'Torvalds', slug: 'torvalds' },
    security: { display_name: 'Diffie', slug: 'diffie' },
    dba: { display_name: 'Codd', slug: 'codd' },
    ux: { display_name: 'Norman', slug: 'norman' },
    'code-quality': { display_name: 'Knuth', slug: 'knuth' },
    data: { display_name: 'Hopper', slug: 'hopper' },
    iac: { display_name: 'Hashimoto', slug: 'hashimoto' },
    tooling: { display_name: 'McCarthy', slug: 'mccarthy' },
  },
  starwars: {
    architect: { display_name: 'Yoda', slug: 'yoda' },
    backend: { display_name: 'Han', slug: 'han' },
    frontend: { display_name: 'Leia', slug: 'leia' },
    qa: { display_name: 'Ahsoka', slug: 'ahsoka' },
    infra: { display_name: 'Chewbacca', slug: 'chewbacca' },
    security: { display_name: 'Vader', slug: 'vader' },
    dba: { display_name: 'R2-D2', slug: 'r2-d2' },
    ux: { display_name: 'Padmé', slug: 'padme' },
    'code-quality': { display_name: 'Obi-Wan', slug: 'obi-wan' },
    data: { display_name: 'C-3PO', slug: 'c-3po' },
    iac: { display_name: 'Rey', slug: 'rey' },
    tooling: { display_name: 'Babu Frik', slug: 'babu-frik' },
  },
  tolkien: {
    architect: { display_name: 'Gandalf', slug: 'gandalf' },
    backend: { display_name: 'Aragorn', slug: 'aragorn' },
    frontend: { display_name: 'Arwen', slug: 'arwen' },
    qa: { display_name: 'Legolas', slug: 'legolas' },
    infra: { display_name: 'Gimli', slug: 'gimli' },
    security: { display_name: 'Boromir', slug: 'boromir' },
    dba: { display_name: 'Elrond', slug: 'elrond' },
    ux: { display_name: 'Galadriel', slug: 'galadriel' },
    'code-quality': { display_name: 'Faramir', slug: 'faramir' },
    data: { display_name: 'Bilbo', slug: 'bilbo' },
    iac: { display_name: 'Aulë', slug: 'aule' },
    tooling: { display_name: 'Celebrimbor', slug: 'celebrimbor' },
  },
  turma: {
    architect: { display_name: 'Franjinha', slug: 'franjinha' },
    backend: { display_name: 'Cebolinha', slug: 'cebolinha' },
    frontend: { display_name: 'Magali', slug: 'magali' },
    qa: { display_name: 'Mônica', slug: 'monica' },
    infra: { display_name: 'Cascão', slug: 'cascao' },
    security: { display_name: 'Bidu', slug: 'bidu' },
    dba: { display_name: 'Marocas', slug: 'marocas' },
    ux: { display_name: 'Anjinho', slug: 'anjinho' },
    'code-quality': { display_name: 'Titi', slug: 'titi' },
    data: { display_name: 'Chico', slug: 'chico' },
    iac: { display_name: 'Piteco', slug: 'piteco' },
    tooling: { display_name: 'Nimbus', slug: 'nimbus' },
  },
  egyptian: {
    architect: { display_name: 'Thoth', slug: 'thoth' },
    backend: { display_name: 'Rá', slug: 'ra' },
    frontend: { display_name: 'Ísis', slug: 'isis' },
    qa: { display_name: 'Hórus', slug: 'horus' },
    infra: { display_name: 'Ptah', slug: 'ptah' },
    security: { display_name: 'Anúbis', slug: 'anubis' },
    dba: { display_name: 'Seshat', slug: 'seshat' },
    ux: { display_name: 'Bastet', slug: 'bastet' },
    'code-quality': { display_name: 'Maat', slug: 'maat' },
    data: { display_name: 'Osíris', slug: 'osiris' },
    iac: { display_name: 'Imhotep', slug: 'imhotep' },
    tooling: { display_name: 'Khnum', slug: 'khnum' },
  },
}

// presetOrder é a ordem canônica e estável em que os nomes de presets são
// listados (ex.: por presetNames e em mensagens de erro).
const PRESET_ORDER = ['greek', 'norse', 'potter', 'thrones', 'chaves', 'pioneers', 'starwars', 'tolkien', 'turma', 'egyptian']

// preset retorna a Config de identidade para o nome de preset temático dado.
// Se name não for um preset conhecido, lança erro listando os nomes válidos.
//
// A Config retornada é sempre uma cópia: mutá-la não afeta a tabela de
// presets no nível do módulo nem qualquer outra Config retornada por uma
// chamada anterior a preset().
function preset(name) {
  const agents = PRESETS[name]
  if (!agents) {
    throw new Error(`identity: preset desconhecido "${name}" (validos: ${presetNames().join(', ')})`)
  }
  const copied = {}
  for (const [id, agent] of Object.entries(agents)) {
    copied[id] = { display_name: agent.display_name, slug: agent.slug }
  }
  return { schema_version: 1, agents: copied }
}

// presetNames retorna os nomes de todos os presets conhecidos, na ordem
// estável: greek, norse, potter, thrones, chaves, pioneers, starwars,
// tolkien, turma, egyptian.
function presetNames() {
  return [...PRESET_ORDER]
}

// Verificação em tempo de carga do módulo: presetOrder e PRESETS devem estar
// em sincronia — guarda defensiva contra drift silencioso entre o mapa e a
// ordem documentada (espelha o init() de internal/identity/preset.go).
;(function assertPresetOrderInSync() {
  if (PRESET_ORDER.length !== Object.keys(PRESETS).length) {
    throw new Error('identity: presetOrder e PRESETS fora de sincronia')
  }
  const seen = new Set()
  for (const name of PRESET_ORDER) {
    if (!PRESETS[name]) {
      throw new Error(`identity: preset "${name}" listado em presetOrder mas ausente de PRESETS`)
    }
    seen.add(name)
  }
  if (seen.size !== PRESET_ORDER.length) {
    throw new Error('identity: presetOrder contem nomes duplicados')
  }
})()

module.exports = { knownAgentIds, preset, presetNames }

File: internal/identity/preset.go
package identity

import (
	"fmt"
	"strings"
)

// KnownAgentIDs returns the canonical list of agent ids known to trackfw, in
// a stable, deterministic order.
func KnownAgentIDs() []string {
	return []string{
		"architect",
		"backend",
		"frontend",
		"qa",
		"infra",
		"security",
		"dba",
		"ux",
		"code-quality",
		"data",
		"iac",
		"tooling",
	}
}

// presets holds the catalog of themed identity presets. DisplayName and Slug
// are HARDCODED here — not derived by calling Slugify at runtime — because
// this is an explicit ADR decision: hardcoding avoids depending on Unicode
// normalization behaving identically across the Go, Node.js and Python CLIs.
var presets = map[string]map[string]AgentIdentity{
	"greek": {
		"architect":    {DisplayName: "Zeus", Slug: "zeus"},
		"backend":      {DisplayName: "Apolo", Slug: "apolo"},
		"frontend":     {DisplayName: "Afrodite", Slug: "afrodite"},
		"qa":           {DisplayName: "Ártemis", Slug: "artemis"},
		"infra":        {DisplayName: "Ares", Slug: "ares"},
		"security":     {DisplayName: "Hades", Slug: "hades"},
		"dba":          {DisplayName: "Poseidon", Slug: "poseidon"},
		"ux":           {DisplayName: "Atena", Slug: "atena"},
		"code-quality": {DisplayName: "Hefesto", Slug: "hefesto"},
		"data":         {DisplayName: "Métis", Slug: "metis"},
		"iac":          {DisplayName: "Dédalo", Slug: "dedalo"},
		"tooling":      {DisplayName: "Prometeu", Slug: "prometeu"},
	},
	"norse": {
		"architect":    {DisplayName: "Odin", Slug: "odin"},
		"backend":      {DisplayName: "Thor", Slug: "thor"},
		"frontend":     {DisplayName: "Freya", Slug: "freya"},
		"qa":           {DisplayName: "Heimdall", Slug: "heimdall"},
		"infra":        {DisplayName: "Tyr", Slug: "tyr"},
		"security":     {DisplayName: "Vidar", Slug: "vidar"},
		"dba":          {DisplayName: "Njord", Slug: "njord"},
		"ux":           {DisplayName: "Idun", Slug: "idun"},
		"code-quality": {DisplayName: "Bragi", Slug: "bragi"},
		"data":         {DisplayName: "Mimir", Slug: "mimir"},
		"iac":          {DisplayName: "Ivaldi", Slug: "ivaldi"},
		"tooling":      {DisplayName: "Loki", Slug: "loki"},
	},
	"potter": {
		"architect":    {DisplayName: "Dumbledore", Slug: "dumbledore"},
		"backend":      {DisplayName: "Snape", Slug: "snape"},
		"frontend":     {DisplayName: "Luna", Slug: "luna"},
		"qa":           {DisplayName: "Moody", Slug: "moody"},
		"infra":        {DisplayName: "Hagrid", Slug: "hagrid"},
		"security":     {DisplayName: "Kingsley", Slug: "kingsley"},
		"dba":          {DisplayName: "Flitwick", Slug: "flitwick"},
		"ux":           {DisplayName: "Tonks", Slug: "tonks"},
		"code-quality": {DisplayName: "Hermione", Slug: "hermione"},
		"data":         {DisplayName: "Trelawney", Slug: "trelawney"},
		"iac":          {DisplayName: "Rowena", Slug: "rowena"},
		"tooling":      {DisplayName: "Ollivander", Slug: "ollivander"},
	},
	"thrones": {
		"architect":    {DisplayName: "Tyrion", Slug: "tyrion"},
		"backend":      {DisplayName: "Jon", Slug: "jon"},
		"frontend":     {DisplayName: "Sansa", Slug: "sansa"},
		"qa":           {DisplayName: "Arya", Slug: "arya"},
		"infra":        {DisplayName: "Brienne", Slug: "brienne"},
		"security":     {DisplayName: "Varys", Slug: "varys"},
		"dba":          {DisplayName: "Samwell", Slug: "samwell"},
		"ux":           {DisplayName: "Margaery", Slug: "margaery"},
		"code-quality": {DisplayName: "Stannis", Slug: "stannis"},
		"data":         {DisplayName: "Bran", Slug: "bran"},
		"iac":          {DisplayName: "Gendry", Slug: "gendry"},
		"tooling":      {DisplayName: "Qyburn", Slug: "qyburn"},
	},
	"chaves": {
		"architect":    {DisplayName: "Girafales", Slug: "girafales"},
		"backend":      {DisplayName: "Madruga", Slug: "madruga"},
		"frontend":     {DisplayName: "Chiquinha", Slug: "chiquinha"},
		"qa":           {DisplayName: "Florinda", Slug: "florinda"},
		"infra":        {DisplayName: "Barriga", Slug: "barriga"},
		"security":     {DisplayName: "Quico", Slug: "quico"},
		"dba":          {DisplayName: "Clotilde", Slug: "clotilde"},
		"ux":           {DisplayName: "Popis", Slug: "popis"},
		"code-quality": {DisplayName: "Nhonho", Slug: "nhonho"},
		"data":         {DisplayName: "Godinez", Slug: "godinez"},
		"iac":          {DisplayName: "Chaves", Slug: "chaves"},
		"tooling":      {DisplayName: "Chapolin", Slug: "chapolin"},
	},
	"pioneers": {
		"architect":    {DisplayName: "Turing", Slug: "turing"},
		"backend":      {DisplayName: "Ritchie", Slug: "ritchie"},
		"frontend":     {DisplayName: "Berners-Lee", Slug: "berners-lee"},
		"qa":           {DisplayName: "Hamilton", Slug: "hamilton"},
		"infra":        {DisplayName: "Torvalds", Slug: "torvalds"},
		"security":     {DisplayName: "Diffie", Slug: "diffie"},
		"dba":          {DisplayName: "Codd", Slug: "codd"},
		"ux":           {DisplayName: "Norman", Slug: "norman"},
		"code-quality": {DisplayName: "Knuth", Slug: "knuth"},
		"data":         {DisplayName: "Hopper", Slug: "hopper"},
		"iac":          {DisplayName: "Hashimoto", Slug: "hashimoto"},
		"tooling":      {DisplayName: "McCarthy", Slug: "mccarthy"},
	},
	"starwars": {
		"architect":    {DisplayName: "Yoda", Slug: "yoda"},
		"backend":      {DisplayName: "Han", Slug: "han"},
		"frontend":     {DisplayName: "Leia", Slug: "leia"},
		"qa":           {DisplayName: "Ahsoka", Slug: "ahsoka"},
		"infra":        {DisplayName: "Chewbacca", Slug: "chewbacca"},
		"security":     {DisplayName: "Vader", Slug: "vader"},
		"dba":          {DisplayName: "R2-D2", Slug: "r2-d2"},
		"ux":           {DisplayName: "Padmé", Slug: "padme"},
		"code-quality": {DisplayName: "Obi-Wan", Slug: "obi-wan"},
		"data":         {DisplayName: "C-3PO", Slug: "c-3po"},
		"iac":          {DisplayName: "Rey", Slug: "rey"},
		"tooling":      {DisplayName: "Babu Frik", Slug: "babu-frik"},
	},
	"tolkien": {
		"architect":    {DisplayName: "Gandalf", Slug: "gandalf"},
		"backend":      {DisplayName: "Aragorn", Slug: "aragorn"},
		"frontend":     {DisplayName: "Arwen", Slug: "arwen"},
		"qa":           {DisplayName: "Legolas", Slug: "legolas"},
		"infra":        {DisplayName: "Gimli", Slug: "gimli"},
		"security":     {DisplayName: "Boromir", Slug: "boromir"},
		"dba":          {DisplayName: "Elrond", Slug: "elrond"},
		"ux":           {DisplayName: "Galadriel", Slug: "galadriel"},
		"code-quality": {DisplayName: "Faramir", Slug: "faramir"},
		"data":         {DisplayName: "Bilbo", Slug: "bilbo"},
		"iac":          {DisplayName: "Aulë", Slug: "aule"},
		"tooling":      {DisplayName: "Celebrimbor", Slug: "celebrimbor"},
	},
	"turma": {
		"architect":    {DisplayName: "Franjinha", Slug: "franjinha"},
		"backend":      {DisplayName: "Cebolinha", Slug: "cebolinha"},
		"frontend":     {DisplayName: "Magali", Slug: "magali"},
		"qa":           {DisplayName: "Mônica", Slug: "monica"},
		"infra":        {DisplayName: "Cascão", Slug: "cascao"},
		"security":     {DisplayName: "Bidu", Slug: "bidu"},
		"dba":          {DisplayName: "Marocas", Slug: "marocas"},
		"ux":           {DisplayName: "Anjinho", Slug: "anjinho"},
		"code-quality": {DisplayName: "Titi", Slug: "titi"},
		"data":         {DisplayName: "Chico", Slug: "chico"},
		"iac":          {DisplayName: "Piteco", Slug: "piteco"},
		"tooling":      {DisplayName: "Nimbus", Slug: "nimbus"},
	},
	"egyptian": {
		"architect":    {DisplayName: "Thoth", Slug: "thoth"},
		"backend":      {DisplayName: "Rá", Slug: "ra"},
		"frontend":     {DisplayName: "Ísis", Slug: "isis"},
		"qa":           {DisplayName: "Hórus", Slug: "horus"},
		"infra":        {DisplayName: "Ptah", Slug: "ptah"},
		"security":     {DisplayName: "Anúbis", Slug: "anubis"},
		"dba":          {DisplayName: "Seshat", Slug: "seshat"},
		"ux":           {DisplayName: "Bastet", Slug: "bastet"},
		"code-quality": {DisplayName: "Maat", Slug: "maat"},
		"data":         {DisplayName: "Osíris", Slug: "osiris"},
		"iac":          {DisplayName: "Imhotep", Slug: "imhotep"},
		"tooling":      {DisplayName: "Khnum", Slug: "khnum"},
	},
}

// presetOrder is the canonical, stable order in which preset names are
// listed (e.g. by PresetNames and in error messages).
var presetOrder = []string{"greek", "norse", "potter", "thrones", "chaves", "pioneers", "starwars", "tolkien", "turma", "egyptian"}

// Preset returns the identity Config for the given themed preset name (one
// of "greek", "norse", "potter", "thrones", "chaves", "pioneers", "starwars",
// "tolkien", "turma", "egyptian"). If name is not a known preset, Preset
// returns an error listing the valid preset names.
//
// The returned Config is always a copy: mutating it does not affect the
// package-level preset table or any other Config returned by a previous
// call to Preset.
func Preset(name string) (Config, error) {
	agents, ok := presets[name]
	if !ok {
		return Config{}, fmt.Errorf("identity: preset desconhecido %q (validos: %s)", name, strings.Join(PresetNames(), ", "))
	}

	copied := make(map[string]AgentIdentity, len(agents))
	for id, agent := range agents {
		copied[id] = agent
	}

	return Config{
		SchemaVersion: schemaVersion,
		Agents:        copied,
	}, nil
}

// PresetNames returns the names of all known presets, in the stable order:
// greek, norse, potter, thrones, chaves, pioneers, starwars, tolkien, turma,
// egyptian.
func PresetNames() []string {
	names := make([]string, len(presetOrder))
	copy(names, presetOrder)
	return names
}

// init verifies at package load time that presetOrder and the presets map
// are kept in sync, and that presetOrder is sorted the way we intend to
// document it (defensive: this is a compile-time-adjacent guard against
// silent drift between the map and the documented order).
func init() {
	if len(presetOrder) != len(presets) {
		panic("identity: presetOrder e presets fora de sincronia")
	}
	seen := make(map[string]bool, len(presetOrder))
	for _, name := range presetOrder {
		if _, ok := presets[name]; !ok {
			panic(fmt.Sprintf("identity: preset %q listado em presetOrder mas ausente de presets", name))
		}
		seen[name] = true
	}
	if len(seen) != len(presetOrder) {
		panic("identity: presetOrder contem nomes duplicados")
	}
}
